account_asset_management/models/account_move.py

```diff
@@ -30,13 +30,18 @@ class AccountMove(models.Model):
     asset_count = fields.Integer(compute="_compute_asset_count")
 
     def _compute_asset_count(self):
-        for rec in self:
-            assets = (
-                self.env["account.asset.line"]
-                .search([("move_id", "=", self.id)])
-                .mapped("asset_id")
+        if len(self) < 100:
```

Comment:
I don't see the benefit of doing this. Why not getting always everything by read_group?

Reply:
They are proofs of concept that I am doing.
The idea would be that, with the normal use, in the self variable only 1 invoice comes and when accessing the form view the lines will be loaded/cached, so it is not necessary to make an additional query with the readgroup.
But if this field is added in a tree view or in an export, it is more optimal to perform the readgroup query and avoid loading all the lines.
I have the doubt of what value because given the previous explanation it would make more sense to make the difference between 1 record and more than 1 record, but perhaps it may also be interesting to consult the possible cache of the lines when in case of few records (for example an export containing header and line data)

But as I said, they are tests to debate in OCA if it is a good idea to take optimization to this extreme (including a recommendation in the guidelines or even creating an abstract method to facilitate it) or we despise the cost of execute an additional query for each data of this type getting fewer lines of code.

Reply:
@sbidoul @rousseldenis @moylop260 @yelizariev 
What you think about this?
https://github.com/OCA/account-financial-tools/pull/1135#discussion_r589810737

Reply:
Any final performance measure? I think it's better to standardize code if no too much difference.

Reply:
As it does not matter too much I will continue using read group.